connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -289,6 +301,38 @@ private void checkAndPutConnectorConfigName(String connectorName, Map<String, St
         return completeOrForwardRequest(cb, path, method, body, null, new IdentityTranslator<T>(), forward);
     }
 
+    private <T> T loadConnectorClass(final String className, final Class<T> type) {
+        try {
+            return type.cast(Class.forName(className).newInstance());
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    // go through config parameters and replace password field value with "*"
+    // this return a new map, instead of making change to the original object
+    private Map<String, String> maskCredentials(Map<String, String> config) {
+        final String connectorClassStr = "connector.class";
+        if (!config.containsKey(connectorClassStr)) {
+            return config;
+        }
+
+        Connector connectorClass = loadConnectorClass(config.get(connectorClassStr), Connector.class);
+        Map<String, ConfigDef.ConfigKey> definedConfigKeys = connectorClass.config().configKeys();
+        Map<String, String> newConfig = new HashMap<>();
+
+        for (Map.Entry<String, String> entry : config.entrySet()) {
+            final String key = entry.getKey();
+            if (definedConfigKeys.containsKey(key) && definedConfigKeys.get(key).type.equals(ConfigDef.Type.PASSWORD)) {
```

Comment:
How about looking up the `ConfigKey` once and then checking if it was found?
```
ConfigDef.ConfigKey configKey = definedConfigKeys.get(key);
if (configKey != null && ConfigDef.Type.Password.equals(configKey.type)) {
    ...
```
Also, note I reversed the comparison to be absolutely sure that there is not a NPE should `type` be null.